src/lib/user/actions/get-my-account.ts

```diff
@@ -0,0 +1,35 @@
+import { MyAccount } from "@/lib/data/my-account-model";
+import { UserActor } from "../user-actor";
+import { dbQuery } from "@/lib/data/db-utils";
+import { UserPii } from "@/lib/data/db-models";
+
+export async function getMyAccount(
+  actor: UserActor,
+): Promise<MyAccount | null> {
+  const { userId, dbPool } = actor.getParams();
+
+  const sql = `
+  SELECT
+    user_id,
+    user_creation_time,
+    user_residency
+  FROM users
+  WHERE user_id = $1
+  `;
+  try {
+    const res = await dbQuery(dbPool, sql, [userId]);
+    const { user_id, user_creation_time, user_residency } = res.rows[0];
```

Comment:
We should check `res.rows.length` here and return `null` when it's zero instead of relying on the exception.

Reply:
Done 04170b200ea3681d2c42e1e5250f8b27120a6c0c